engine/src/main/java/com/datatorrent/stram/plan/physical/StreamMapping.java

```diff
@@ -281,6 +282,17 @@ private void redoMapping()
             (sourceSingleFinal != null ? sourceSingleFinal.booleanValue() : PortContext.UNIFIER_SINGLE_FINAL.defaultValue);
 
         if (upstream.size() > 1) {
+          // detach downstream from upstream operator for the case where no unifier existed previously
+          for (PTOutput unifierSource : unifierSources) {
+            Iterator<PTInput> sinks = unifierSource.sinks.iterator();
+            while (sinks.hasNext()) {
+              PTInput sink = sinks.next();
+              if (sink.target == doperEntry.first) {
+                doperEntry.first.inputs.remove(sink);
```

Comment:
I think doperEntry.first needs to be added into undeploy and deploy list. Because, this fix changing the port mappings of existing (doperEntry.first) partition. Please correct it, if I am wrong.